pyccel/codegen/printing/pycode.py

```diff
@@ -884,10 +884,12 @@ def _print_Slice(self, expr):
                 stop  = stop,
                 step  = step)
 
-    def _print_SetAdd(self, expr):
-        name = self._print(expr.set_variable)
-        args = self._print(expr.add_argument)
-        return f"{name}.add({args})\n"
+    def _print_SetMethod(self, expr):
+        set_var = self._print(expr.set_variable)
+        name = self._print(expr.name)
+        args = "" if len(expr.args) == 0 or expr.args[0] is None \
+            else ', '.join(self._print(a) for a in expr.args)
```

Comment:
What happens if `expr.args[0] is None` and `expr.args[1] is not None`? As is this handles all cases that you have so far but I don't know if it handles everything that is possible?

Reply:
Could you please provide me with an example?

Reply:
```python
a = {1,2,3}
a.my_imaginary_set_function_added_in_python_3_20(None, 2)
```
Or:
```python
a = {1,2,3}
a.my_imaginary_set_function_with_args_x_and_y(y=2)
```

Reply:
I don't understand the example you provided. The point is that there isn't a method that can take more than one argument, and sets are homogeneous.

Reply:
There isn't **currently** a method that can take more than one argument. But I would rather write future proof code.

> and sets are homogeneous.

I'm not sure why this is relevant.

For example, let's imagine that they add a `insert_range` method to `set` with the following prototype:
```python
def insert_range(self, *, start = None, stop, step = None):
     start = start or 0
     step = step or 1
```
I can call this as:
```python
a.insert_range(stop=4)
```
when you print you would have `len(args)==3`, and `args[0] is None`

Reply:
Why is args[0] None if the start will default to 0?

Reply:
That depends on how you implement it. Inside it defaults to 0, but the argument that is passed is still None. Remember this is a contrived example to try and explain how you can have multiple arguments with `args[0] is None` it is not something that anyone plans to add to Python. I just want to make sure that your implementation is general enough.

Reply:
Yeah, I understand your point, but I'm not seeing any scenario where this is necessary to consider for a solution. If you could offer a better solution, that would be helpful. 
If a method with a similar idea exists, I believe the arguments should be handled similarly to the` range class`:"
```Python
 n = len(args)

        if n == 1:
            self._start = LiteralInteger(0)
            self._stop  = args[0]
            self._step  = LiteralInteger(1)
        elif n == 2:
            self._start = args[0]
            self._stop  = args[1]
            self._step  = LiteralInteger(1)
        elif n == 3:
            self._start = args[0]
            self._stop  = args[1]
            self._step  = args[2]
```

Reply:
The simplest replacement is:
```suggestion
        args = "" if len(expr.args) == 0 or expr.args[-1] is None \
            else ', '.join(self._print(a) for a in expr.args)
```
This will have the same result but won't lead to odd results if there are multiple args

Reply:
fixed : 0c483df

Reply:
[eb9da2e](https://github.com/pyccel/pyccel/pull/1764/commits/eb9da2ebcce54acf8cf196c04a497d47c9ceda11)